### DISCUSSION TITLE
Can I get hierarchical relations tree ?

### DISCUSSION THREAD

**Discussion by Ben-Duf** [Gremlinq/ExRam.Gremlinq#799] [Q&A]
In gremlin I can travers to children hierarchies through below "repeat - out - tree" query and get hierarchical (tree) infos (not just all vertex). Is there anything available with Gremlinq?

What I actually have :
```
            var query = _g
                .V<ScopeVertex>(scopeId, pk)
                .Loop(_ => _.Emit().Repeat(__ => __.OutE<ChildMemberEdge>().InV<ScopeVertex>())
```
which generates :
`g.V('scopeId','pk').hasLabel('ScopeVertex').repeat(out('ChildMemberEdge').hasLabel('ScopeVertex')).emit()`
but this returns only vertexes, with no hierarchy info.

=> what I need is vertexes with "parent Id" or use the ".tree()" step
`g.V('scopeId','pk').hasLabel('ScopeVertex').repeat(out('ChildMemberEdge').hasLabel('ScopeVertex')).emit().tree()`
but I don't find how to do that with gremlinQ

can anyone help please ?

Thanks.

**Comment by jayaemekar** [comment]
Are you able to resolve this? Can you please share the solution
